src/microbe_stage/editor/MicrobePartSelection.cs

```diff
@@ -105,11 +120,20 @@ public override void _Ready()
         iconRect = GetNode<TextureRect>("VBoxContainer/Button/Icon");
         nameLabel = GetNode<Label>("VBoxContainer/Name");
 
+        OnDisplayPartNamesChanged(Settings.Instance.DisplayPartNames);
+        Settings.Instance.DisplayPartNames.OnChanged += OnDisplayPartNamesChanged;
```

Comment:
This needs to unregister. So you need add an exit tree method override that unregisters this.